### ISSUE TITLE
Experimenting with branch imswitch-core

### ISSUE THREAD

**Issue by xcasas** [kasasxav/ImSwitch#211]
Hi,

I opened a branch [imswitch-core](https://github.com/ImSwitch/ImSwitch/tree/imswitch-core) to explore the concept that @jacopoabramo and I [have been thinking](https://github.com/ImSwitch/ImSwitch/discussions/210), a minimal ImSwitch to interact with ImTools. The widgets, widgetcontrollers and managers would be added dynamically using a plugin system.

**Comment by beniroquai** [comment]
New Plugin system or based on what's already there?

**Comment by jacopoabramo** [comment]
@beniroquai it'll be based on what is currently existing. we'll have to work to adapt it in such a way that we can do the following:

- the core will contain the essential parts needed by imswitch to operate; ideally this should be enough to launch imswitch in a sort of empty state: no widgets and controllers, only things like the communication channel and the master controller;
- the toolkit will contain the parts needed to develop plugins in the following format:
  - widget-controller pairs;
  - single managers;
  - full stack of widget-controller-manager

We discussed that widget-controller pairs and managers that were developed as part of the TestaLab group should be moved to a separate repository as a plugin "bundle", licensing them in a way that is compliant to the wishes of the lab members. Same thing goes for the UC2 widget-controller pairs and managers should be moved to a separate repo as a plugin "bundle", so that you can get full credit of those and you can license them however you wish.

We also discussed the possibility that other modules such as the `imreconstruct` and the `imscripting` should be treated as plugins as well but with a more complex structure; the reason is due to the fact that those two modules were developed as a specific need for the super resolution group from TestaLab, thus making them too specific to consider them as core modules.

**Comment by kasasxav** [comment]
Exactly! We don't only want the plugins that are widget+widgetController, but also the possibility to have one repo that is a plugin, and it's composed by many widgets, widgetControllers, and managers.

the idea of the modules as plugins is also that if some developer wants to release a new module that everyone could use (like SIM reconstruction), they can also do it as a plugin and release it open source

**Comment by beniroquai** [comment]
Sure.
We should not loose traction before the transition is done and the modules are actually moved to the separate repositories, the knowledge is transferred to all developers and the idea is not dying on the way to perfection 😉🥲 

I'll do what ever you want 😇
